crates/uv/tests/run.rs

```diff
@@ -1950,3 +1950,44 @@ fn run_invalid_project_table() -> Result<()> {
 
     Ok(())
 }
+
+#[test]
+/// Check warning message for https://github.com/astral-sh/uv/issues/6998.
+fn run_script_without_build_system() -> Result<()> {
+    let context = TestContext::new("3.12");
+
+    let pyproject_toml = context.temp_dir.child("pyproject.toml");
+    pyproject_toml.write_str(indoc! { r#"
+        [project]
+        name = "foo"
+        version = "0.1.0"
+        requires-python = ">=3.12"
+        dependencies = []
+
+        [project.scripts]
+        entry = "foo:custom_entry"
+        "#
+    })?;
+
+    let test_script = context.temp_dir.child("src/__init__.py");
+    test_script.write_str(indoc! { r#"
+        def custom_entry():
+            print!("Hello")
+       "#
+    })?;
+
+    uv_snapshot!(context.filters(), context.run().arg("entry"), @r###"
+    success: false
+    exit_code: 2
+    ----- stdout -----
+
+    ----- stderr -----
+    warning: No `build-system` section found for the package; consider adding one to correctly build the project
```

Comment:
This is now a
```
Skipping scripts installation because this project is not a package. To install them, consider setting `tool.uv.package = true` or configuring a custom `build-system.`
```